File: java/Samples/Biometrics/VB/AbisSampleVB/SubjectEditor/EditPhrasesForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Forms

Partial Public Class EditPhrasesForm
	Inherits Form
	#Region "Private fields"

	Private _phrases As List(Of Phrase)

	#End Region

	#Region "Public constructor"

	Public Sub New()
		InitializeComponent()
	End Sub

	#End Region

	#Region "Public properties"

	Public Property Phrases() As List(Of Phrase)
		Get
			Return _phrases
		End Get
		Set(ByVal value As List(Of Phrase))
			_phrases = value
			ListAllPhrases()
		End Set
	End Property

	#End Region

	#Region "Private form events"

	Private Sub BtnAddClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnAdd.Click
		Dim phraseId As Integer

		If String.IsNullOrEmpty(tbPhraseId.Text) OrElse String.IsNullOrEmpty(tbPhrase.Text) Then
			Utilities.ShowInformation("One or more fields is empty. Please fill all of the fields.")
			Return
		End If

		If (Not Integer.TryParse(tbPhraseId.Text, phraseId)) Then
			Utilities.ShowError("Phrase Id should be integer (above 0)!")
			Return
		End If

		If phraseId <= 0 Then
			Utilities.ShowError("Phrase Id should be above 0!")
			Return
		End If

		If tbPhrase.Text.Contains("=") OrElse tbPhrase.Text.Contains(";") Then
			Utilities.ShowError("Phrase must not countain symbols: '=', ';'!")
			Return
		End If

		Dim phraseToAdd As New Phrase(phraseId, tbPhrase.Text)
		For Each phrase As Phrase In _phrases
			If phrase.Id = phraseToAdd.Id Then
				Utilities.ShowError("Another phrase with the same phrase id already exist in the list!")
				Return
			End If
		Next phrase

		_phrases.Add(phraseToAdd)
		ListAllPhrases()
	End Sub

	Private Sub BtnRemoveClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnRemove.Click
		If lvPhrases.SelectedItems.Count = 1 Then
			Dim toRemove As ListViewItem = lvPhrases.SelectedItems(0)
			Dim phrase As Phrase = TryCast(toRemove.Tag, Phrase)
			If phrase IsNot Nothing Then
				_phrases.Remove(phrase)
			End If
			lvPhrases.Items.Remove(toRemove)
		Else
			Utilities.ShowInformation("No or more than one phrases are selected!")
		End If
	End Sub

	Private Sub BtnCloseClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
		Close()
	End Sub

	#End Region

	#Region "Private methods"

	Private Sub ListAllPhrases()
		lvPhrases.Items.Clear()
		For Each phrase As Phrase In _phrases
			Dim item As New ListViewItem(New String() { phrase.Id.ToString(), phrase.String })
			item.Tag = phrase
			lvPhrases.Items.Add(item)
		Next phrase
		If lvPhrases.Items.Count > 0 Then
			lvPhrases.SelectedIndices.Add(0)
		End If
	End Sub

	#End Region
End Class

File: java/Samples/Biometrics/CS/AbisSampleCS/SubjectEditor/EditPhrasesForm.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;

namespace Neurotec.Samples
{
	public partial class EditPhrasesForm : Form
	{
		#region Private fields

		private List<Phrase> _phrases;

		#endregion

		#region Public constructor

		public EditPhrasesForm()
		{
			InitializeComponent();
		}

		#endregion

		#region Public properties

		public List<Phrase> Phrases
		{
			get { return _phrases; }
			set
			{
				_phrases = value;
				ListAllPhrases();
			}
		}

		#endregion

		#region Private form events

		private void BtnAddClick(object sender, EventArgs e)
		{
			int phraseId;

			if (string.IsNullOrEmpty(tbPhraseId.Text) || string.IsNullOrEmpty(tbPhrase.Text))
			{
				Utilities.ShowInformation("One or more fields is empty. Please fill all of the fields.");
				return;
			}

			if (!int.TryParse(tbPhraseId.Text, out  phraseId))
			{
				Utilities.ShowError("Phrase Id should be integer (above 0)!");
				return;
			}

			if (phraseId <= 0)
			{
				Utilities.ShowError("Phrase Id should be above 0!");
				return;
			}

			if (tbPhrase.Text.Contains("=") || tbPhrase.Text.Contains(";"))
			{
				Utilities.ShowError("Phrase must not countain symbols: '=', ';'!");
				return;
			}

			Phrase phraseToAdd = new Phrase(phraseId, tbPhrase.Text);
			foreach (Phrase phrase in _phrases)
			{
				if (phrase.Id == phraseToAdd.Id)
				{
					Utilities.ShowError("Another phrase with the same phrase id already exist in the list!");
					return;
				}
			}

			_phrases.Add(phraseToAdd);
			ListAllPhrases();
		}

		private void BtnRemoveClick(object sender, EventArgs e)
		{
			if (lvPhrases.SelectedItems.Count == 1)
			{
				ListViewItem toRemove = lvPhrases.SelectedItems[0];
				Phrase phrase = toRemove.Tag as Phrase;
				if (phrase != null)
					_phrases.Remove(phrase);
				lvPhrases.Items.Remove(toRemove);
			}
			else
			{
				Utilities.ShowInformation("No or more than one phrases are selected!");
			}
		}

		private void BtnCloseClick(object sender, EventArgs e)
		{
			Close();
		}

		#endregion

		#region Private methods

		private void ListAllPhrases()
		{
			lvPhrases.Items.Clear();
			foreach (Phrase phrase in _phrases)
			{
				ListViewItem item = new ListViewItem(new string[] { phrase.Id.ToString(), phrase.String });
				item.Tag = phrase;
				lvPhrases.Items.Add(item);
			}
			if (lvPhrases.Items.Count > 0)
				lvPhrases.SelectedIndices.Add(0);
		}

		#endregion
	}
}
